pkg/querier/blocks_store_queryable_test.go

```diff
@@ -893,50 +893,64 @@ func TestBlocksStoreQuerier_Select(t *testing.T) {
+	}{
+		"canceled request on series stream": {
+			produceSeries: false,
+		},
+		"canceled request on receiving series stream": {
+			produceSeries: true,
+		},
+	}
 
-		ctx = limiter.AddQueryLimiterToContext(ctx, noOpQueryLimiter)
-		reg := prometheus.NewPedanticRegistry()
+	for testName, testData := range canceledRequestTests {
+		t.Run(testName, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
 
-		storeGateway := &cancelerStoreGatewayClientMock{
-			remoteAddr: "1.1.1.1",
-			cancel:     cancel,
-		}
+			ctx = limiter.AddQueryLimiterToContext(ctx, noOpQueryLimiter)
+			reg := prometheus.NewPedanticRegistry()
 
-		stores := &blocksStoreSetMock{mockedResponses: []interface{}{
-			map[BlocksStoreClient][]ulid.ULID{storeGateway: {block1}},
-			errors.New("no store-gateway remaining after exclude"),
-		}}
-
-		finder := &blocksFinderMock{}
-		finder.On("GetBlocks", mock.Anything, "user-1", minT, maxT).Return(bucketindex.Blocks{
-			{ID: block1},
-		}, map[ulid.ULID]*bucketindex.BlockDeletionMark(nil), nil)
-
-		q := &blocksStoreQuerier{
-			ctx:         ctx,
-			minT:        minT,
-			maxT:        maxT,
-			userID:      "user-1",
-			finder:      finder,
-			stores:      stores,
-			consistency: NewBlocksConsistencyChecker(0, 0, log.NewNopLogger(), nil),
-			logger:      log.NewNopLogger(),
-			metrics:     newBlocksStoreQueryableMetrics(reg),
-			limits:      &blocksStoreLimitsMock{},
-		}
+			storeGateway := &cancelerStoreGatewayClientMock{
+				remoteAddr:    "1.1.1.1",
+				produceSeries: testData.produceSeries,
+				cancel:        cancel,
+			}
 
-		matchers := []*labels.Matcher{
-			labels.MustNewMatcher(labels.MatchEqual, labels.MetricName, metricName),
-		}
+			stores := &blocksStoreSetMock{mockedResponses: []interface{}{
+				map[BlocksStoreClient][]ulid.ULID{storeGateway: {block1}},
+				errors.New("no store-gateway remaining after exclude"),
+			}}
 
-		sp := &storage.SelectHints{Start: minT, End: maxT}
-		set := q.Select(true, sp, matchers...)
-		require.Error(t, set.Err())
-		require.ErrorIs(t, set.Err(), context.Canceled)
-	})
+			finder := &blocksFinderMock{}
+			finder.On("GetBlocks", mock.Anything, "user-1", minT, maxT).Return(bucketindex.Blocks{
+				{ID: block1},
+			}, map[ulid.ULID]*bucketindex.BlockDeletionMark(nil), nil)
+
+			q := &blocksStoreQuerier{
+				ctx:         ctx,
+				minT:        minT,
+				maxT:        maxT,
+				userID:      "user-1",
+				finder:      finder,
+				stores:      stores,
+				consistency: NewBlocksConsistencyChecker(0, 0, log.NewNopLogger(), nil),
+				logger:      log.NewNopLogger(),
+				metrics:     newBlocksStoreQueryableMetrics(reg),
+				limits:      &blocksStoreLimitsMock{},
+			}
+
+			matchers := []*labels.Matcher{
+				labels.MustNewMatcher(labels.MatchEqual, labels.MetricName, metricName),
+			}
+
+			sp := &storage.SelectHints{Start: minT, End: maxT}
+			set := q.Select(true, sp, matchers...)
+			require.Error(t, set.Err())
```

Comment:
```suggestion
```

Just `ErrorIs` should be sufficient, no?

Reply:
i suspect you left this out because it was already like this before your change? fine by me to leave it as-is